fred2/ignoreordersdlg.h

```diff
@@ -30,8 +30,8 @@ class ignore_orders_dlg : public CDialog
 	ignore_orders_dlg(CWnd* pParent = NULL);   // standard constructor
 
 	ship *m_shipp;
-	check_box_info check_boxes[MAX_CHECKBOXES];
-	int m_num_checks_active;
+	SCP_vector<size_t> m_orderList;
+	std::unique_ptr<CCheckListBox> m_checklistbox;
```

Comment:
Is it possible to declare `m_checklistbox` as part of the object *without* using dynamic memory allocation? I think that could make the changes a good bit simpler.

Take a look at how `CAddModifyContainerDlg` does this with `m_container_data_lister`.

Reply:
Also, a name for this variable that indicates what the check boxes mean, as `m_container_data_lister` does,, would be really nice.